packages/astro/src/core/app/index.ts

```diff
@@ -307,6 +317,9 @@ export class App {
 		if (!routeData) {
 			return this.#renderError(request, { locals, status: 404 });
 		}
+		if (getEnv) {
+			setGetEnv(getEnv);
```

Comment:
Shouldn't we do this only if the experimental flag is turned on? In order to retrieve the experimental flag from SSR, you'll have to add a new property to the `SSRManifest` type